libxrdp/xrdp_sec.c

```diff
@@ -2252,7 +2252,11 @@ xrdp_sec_incoming(struct xrdp_sec *self)
     }
 
     /* initialize selected security layer */
-    if (iso->selectedProtocol > PROTOCOL_RDP)
+    if (self->rdp_layer->client_info.vmconnect)
+    {
+        /* Security handled by host: do nothing. */
```

Comment:
The important part is this, we just skip entirely the security part.

Reply:
We can probably make this dependent on running over a vsock. What do you think?